crypto/src/key/hdkd/derivation_path.rs

```diff
@@ -134,6 +151,20 @@ mod tests {
     use super::*;
     use test_utils::{assert_encoded_eq, decode_from_hex};
 
+    #[test]
+    fn path_diff() {
```

Comment:
Please, without me having to ask for this, tests can go in two ways:

1. Exhaustive manually (something like parse_derivation_path() test below)
2. Exhaustive through randomization/propery-based testing, like most of the tests in mintlayer-core.

I'm frustrated.

Reply:
To be clear, this isn't just bad because it breaks the bylaws of the code, but it also omits success/error cases + omits corner cases (empty vecs, etc).